File: mavsim_matlab/message_types/msg_autopilot.m
% msg_autopilot
%   - message type for commands sent to the autopilot
%
% mavsim_matlab 
%     - Beard & McLain, PUP, 2012
%     - Update history:  
%         2/12/2019 - RWB
classdef msg_autopilot
   %--------------------------------
    properties
        airspeed_command
        course_command
        altitude_command
        phi_feedforward
    end
    %--------------------------------
    methods
        %------constructor-----------
        function self = msg_autopilot()
            self.airspeed_command = 0;
            self.course_command = 0;
            self.altitude_command = 0;
            self.phi_feedforward = 0;
        end
    end
end

File: mavsim_python/message_types/msg_autopilot.py
"""
msg_autopilot
    - messages type for input to the autopilot
    
part of mavsim_python
    - Beard & McLain, PUP, 2012
    - Last update:
        2/5/2019 - RWB
"""


class MsgAutopilot:
    def __init__(self):
        self.airspeed_command = 0.0  # commanded airspeed m/s
        self.course_command = 0.0  # commanded course angle in rad
        self.altitude_command = 0.0  # commanded altitude in m
        self.phi_feedforward = 0.0  # feedforward command for roll angle
